ChangeLog.md

```diff
@@ -19,6 +19,22 @@ See docs/process.md for how version tagging works.
 Current Trunk
 -------------
 
+ - Add a new system for managing system libraries. (#8780)
+   This may require minor changes when performing certain operations:
+     - When using `embuilder.py` to build a specific library, the name may have
+       changed: for consistency, all library names are prefixed with lib now.
+     - `embuilder.py` now only builds the requested library, and not its dependencies
+       and certain system libraries that are always built. For example, running
+       `embuilder.py build libc` no longer builds `libcompiler_rt` if it hasn't be built.
+     - When using `EMCC_FORCE_STDLIBS` with a list of libraries, you must now use
+       the simplified names, for example, `libmalloc` and `libpthreads` instead of
+       `libdlmalloc` or `libpthreads_stub`. These names will link in the correct
+       version of the library: if the build is configured to use `emmalloc`, `libmalloc`
+       will mean `libemmalloc`, and if thread support is disabled, `libpthreads` will
+       mean `libpthreads_stub`. This allows you to say `libmalloc` or `libpthreads` without
+       worrying about which implementation is supposed to be used, and avoid duplicate
+       symbols if you used the wrong implementation.
```

Comment:
The above comment only pertains to `EMCC_FORCE_STDLIBS` and not to `embuilder.py` command line usage, right?

One core use case of embuilder.py is to drive a prebuild of specific system libraries at compiler toolchain package authoring time, so that these system libraries will not be attempted to be built on demand: this helps give fast first time experience to users, e.g. with Unity when they export their first HTML5 project, on a weak laptop building the system libraries would add several minutes to first build - leading users to perceive that web exports are really slow. Also another reason is that installation is done in 'Program Files' folder, where everything is read-only from the get-go. Building content on demand inside 'Program Files' would not work.

To achieve the above, we use embuilder.py to prebuild libemmalloc and libdlmalloc for example, so that still works? (Briefly looking it appears to me that it does, but just verifying)

Reply:
Yes, `embuilder.py` is meant to be able to build any version of the library you want. `EMCC_FORCE_STDLIBS` now uses the right variant for the current build.